docs/reference/ingest/apis/enrich/put-enrich-policy.asciidoc

```diff
@@ -144,3 +150,200 @@ to documents in the enrich index.
+try the following example.
+
+Use the <<indices-create-index, create index API>>
+to create a source index.
+The field mappings for the source index
+must contain:
+
+* A <<geo-shape,geoshape>> field
+  which the enrich processor can use to match incoming documents
+* One or more enrich fields
+  you'd like to append to incoming documents
+
+[source,console]
+----
+PUT /postal_codes
+{
+    "mappings": {
+        "properties": {
+            "location": {
+                "type": "geo_shape"
+            },
+            "postal_code": {
+                "type": "keyword"
+            } 
+        }
+    }
+}
+----
+
+Use the <<docs-index_,index API>>
+to index data to this source index.
+
+[source,console]
+----
+PUT /postal_codes/_doc/1?refresh=wait_for
+{
+    "location": {
+        "type": "envelope",
+        "coordinates": [[13.0, 53.0], [14.0, 52.0]]
+    },
+    "postal_code": "96598"
+}
+----
+// TEST[continued]
+
+Use the put enrich policy API
+to create an enrich policy
+with the `geo_match` policy type.
+This policy must include:
+
+* One or more source indices
+* A `match_field`,
+  the geoshape field from the source indices
+  used to match incoming documents
+* Enrich fields from the source indices
+  you'd like to append to incoming documents
+
+[source,console]
+----
+PUT /_enrich/policy/postal_policy
+{
+    "geo_match": {
+        "indices": "postal_codes",
+        "match_field": "location",
+        "enrich_fields": ["location","postal_code"]
+    }
+}
+----
+// TEST[continued]
+
+Use the <<execute-enrich-policy-api,execute enrich policy API>>
+to create an enrich index for the policy.
+
+include::execute-enrich-policy.asciidoc[tag=execute-enrich-policy-def]
+
+[source,console]
+----
+POST /_enrich/policy/postal_policy/_execute
+----
+// TEST[continued]
+
+Use the <<put-pipeline-api,put pipeline API>>
+to create an ingest pipeline.
+In the pipeline,
+add an <<enrich-processor,enrich processor>>
+that includes:
+
+* Your enrich policy
+* The `field` of incoming documents used
+  to match the geoshape of documents from the enrich index.
+* The `target_field` used
+  to store appended enrich data for incoming documents.
+* The `shape_relation`,
+ which indicates how the processor matches geoshapes in incoming documents
+ to geoshapes in documents from the enrich index.
+
+[source,console]
+----
+PUT /_ingest/pipeline/postal_lookup
+{
+  "description": "Enrich postal codes",
+  "processors": [
+    {
+      "enrich": {
+        "policy_name": "postal_policy",
+        "field": "geo_location",
+        "target_field": "geo_data",
+        "shape_relation": "INTERSECTS"
+      }
+    }
+  ]
+}
+----
+// TEST[continued]
+
+Use the ingest pipeline
+to index a document.
+The incoming document
+should include the `field`
+specified in your enrich processor.
+
+[source,console]
+----
+PUT /users/_doc/0?pipeline=postal_lookup
+{
+    "first_name": "Mardy",
+    "last_name": "Brown",
+    "geo_location": "POINT (13.5 52.5)"
```

Comment:
I had to use the WKT format here.
When I attempted to use the Elasticsearch format, I got a parsing error.
```
    "geo_location": {
        "type": "point",
        "coordinates": [13.5, 52.5]
    }
```

If this is intended, I can document the limitation.

Reply:
the important thing to note here is that the `geo_point` mapping type does not actually support GeoJSON, like you have it in that snippet. The processor uses the same parsing as is done for that mapping type, so this format is not supported. The `geo_point` mapping docs show a bullet list of all the supported formats. Pretty much everything except geojson :)

There is an open discussion going on about adding support for this soon, but nothing crystalized just yet.

Reply:
Ah. That makes sense. Thanks for the clarification @talevy.